mhkit/tests/power/test_power.py

```diff
@@ -42,66 +43,130 @@ def setUpClass(self):
+            hsg)  # had to just put a random rated current in here
+        
+        self.assertAlmostEqual(TCHD.values[0], self.thcd)
 
+    def test_TCHD_sine_wave_xarray(self):
+        harmonics = xr.Dataset(data_vars={'harmonics':(['index'],self.harmonics_vals)},
+                                coords={'index':self.harmonics_int})
+        hsg = power.quality.harmonic_subgroups(harmonics, self.frequency)
         TCHD = power.quality.total_harmonic_current_distortion(
-            hsg, 18.8)  # had to just put a random rated current in here
+            hsg)  # had to just put a random rated current in here
+        
         self.assertAlmostEqual(TCHD.values[0], self.thcd)
 
-    def test_interharmonics_sine_wave(self):
-        current = pd.Series(self.signal, index=self.samples)
+    def test_interharmonics_sine_wave_pandas(self):
         harmonics = pd.DataFrame(self.harmonics_vals,
-                                 index=self.harmonics_int)
+                                  index=self.harmonics_int)
+        inter_harmonics = power.quality.interharmonics(
+            harmonics, self.frequency)
 
+        for i, j in zip(inter_harmonics.values, self.interharmonic):
+            self.assertAlmostEqual(i[0], j, 1)
+            
+    def test_interharmonics_sine_wave_xarray(self):
+        harmonics = xr.Dataset(data_vars={'harmonics':(['index'],self.harmonics_vals)},
+                                coords={'index':self.harmonics_int})
         inter_harmonics = power.quality.interharmonics(
             harmonics, self.frequency)
 
         for i, j in zip(inter_harmonics.values, self.interharmonic):
             self.assertAlmostEqual(i[0], j, 1)
 
-    def test_instfreq(self):
+    def test_instfreq_pandas(self):
         um = pd.Series(self.signal, index=self.samples)
 
         freq = power.characteristics.instantaneous_frequency(um)
         for i in freq.values:
             self.assertAlmostEqual(i[0], self.frequency, 1)
 
-    def test_dc_power_DataFrame(self):
-        current = pd.DataFrame(self.current_data, columns=['A1', 'A2', 'A3'])
-        voltage = pd.DataFrame(self.voltage_data, columns=['V1', 'V2', 'V3'])
+    def test_instfreq_xarray(self):
+        um = pd.Series(self.signal, index=self.samples)
+        um = um.to_xarray()
+
+        freq = power.characteristics.instantaneous_frequency(um)
+        for i in freq.values:
+            self.assertAlmostEqual(i[0], self.frequency, 1)
+
+    def test_dc_power_pandas(self):
+        current = pd.DataFrame(self.current_data, columns=['A1','A2','A3'])
+        voltage = pd.DataFrame(self.voltage_data, columns=['V1','V2','V3'])
         P = power.characteristics.dc_power(voltage, current)
-        self.assertEqual(P.sum()['Gross'],
-                         (voltage.values * current.values).sum())
+        self.assertEqual(P.sum()['Gross'], 584)
+        P = power.characteristics.dc_power(voltage['V1'], current['A1'])
+        self.assertEqual(P.sum()['Gross'], 70)
```

Comment:
It looks like you changed this from recalculating the values to hardcoding them. 

Are we getting the same values here as before or could you just comment on this change in general?

Reply:
These are exactly the same values that the test should give, just hard coded in. It's been a few weeks, I think I was having issues duplicating the recalculation method easily across pandas and xarray because of their different formats. Looking at it again, I think a better solution is to recalculate the power values, but with self.current_data and self.voltage_data. That would eliminate any differences between xarray/pandas and be easily repeatable across both tests